frontend/src/metabase/lib/errors/console.js

```diff
@@ -0,0 +1,16 @@
+/* eslint-disable no-console */
+export const MAX_ERROR_LOGS = 20;
+
+export function captureConsoleErrors() {
+  console.errorBuffer = [];
+
+  const originalError = console.error;
+
+  console.error = function (...args) {
+    if (console.errorBuffer.length >= MAX_ERROR_LOGS) {
+      console.errorBuffer.pop();
+    }
+    console.errorBuffer.unshift(Array.from(args));
+    originalError(...args);
+  };
+}
```

Comment:
this feels hacky, but I love how simple it is, and I can't think of a better way to do it  🤷‍♀️

Reply:
I dig it.

Would it be useful to also capture console.warn, console.debug and console.trace?

Reply:
You might have thought of this already but another way would be to introduce a new function `logError` which writes to the error buffer and also does `console.error`. Then globally replace all console.error calls with `logError` and ban new usages of console.error in eslint. That's a little more explicit, since it avoids quietly overriding a global, but I think what you have works great.

Reply:
that would be nice, but most of the errors we'll be capturing won't be our own console.errors, it'll be stuff generated by react.